rir/src/interpreter/interp.cpp

```diff
@@ -2188,15 +2188,19 @@ SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP env,
 
         INSTRUCTION(inc_) {
             SEXP val = ostack_top(ctx);
-            assert(TYPEOF(val) == INTSXP);
-            int i = INTEGER(val)[0];
-            if (NO_REFERENCES(val)) {
-                INTEGER(val)[0]++;
-            } else {
+            SLOWASSERT(TYPEOF(val) == INTSXP);
+            // Inc_ destructively modifies TOS, even if the refcount is 1. This
+            // can be used to perform `++i` on a value on the stack. The old i
+            // value will be overwritten (generally only do this if you are sure
+            // that this is the last copy on the stack).
```

Comment:
I don't think this is necessary because you mention the same in `insns.h`. Otherwise, also mention for `subassign`

Reply:
I think this is *very* neccessary, since we both tried to "fix" this